From a7cfe5e7b86ebba4aa184b2612c4ecb70a01aa02 Mon Sep 17 00:00:00 2001
From: "Satyanarayana, Sandhya" <sandhya.satyanarayana@ti.com>
Date: Tue, 22 May 2012 12:33:20 +0530
Subject: [PATCH 05/18] ARM: OMAP: AM33XX: PM: Wait correctly for the PLLs to
 relock in the sleep code

The PLL relock code was incorrectly exiting the loop
before the PLLs relock. Since the PLLs relock within
a couple of A8 assembly instruction execution this went
unnoticed till now. The issue was noticed when validating
the abort path of the suspend process wherein the
incorrect check for PLL relock resulted in a lockup.

Signed-off-by: Satyanarayana, Sandhya <sandhya.satyanarayana@ti.com>
Signed-off-by: Vaibhav Bedia <vaibhav.bedia@ti.com>
---
 arch/arm/mach-omap2/sleep33xx.S |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/sleep33xx.S b/arch/arm/mach-omap2/sleep33xx.S
index b7a1612..79a9e39 100644
--- a/arch/arm/mach-omap2/sleep33xx.S
+++ b/arch/arm/mach-omap2/sleep33xx.S
@@ -61,8 +61,8 @@ pll_lock_\name:
 	ldr	r0, \idlest_addr
 wait_pll_lock_\name:
 	ldr	r1, [r0]
-	tst	r1, #0x1
-	bne	wait_pll_lock_\name
+	ands	r1, #0x1
+	beq	wait_pll_lock_\name
 	.endm
 
 	/* EMIF config for low power mode */
-- 
1.7.7.6

